lightyear/src/utils/ready_buffer.rs

```diff
@@ -160,23 +166,23 @@ pub struct ItemWithReadyKey<K, T> {
     pub item: T,
 }
 
-impl<K: Ord, T: PartialEq> Eq for ItemWithReadyKey<K, T> {}
+impl<K: Ord, T> Eq for ItemWithReadyKey<K, T> {}
 
-impl<K: Ord, T: PartialEq> PartialEq<Self> for ItemWithReadyKey<K, T> {
+impl<K: Ord, T> PartialEq<Self> for ItemWithReadyKey<K, T> {
     fn eq(&self, other: &Self) -> bool {
-        self.item == other.item && self.key == other.key
+        self.key == other.key
```

Comment:
But what if we compare we have two component histories with values at the same ticks but the values are different?
Then we would get `ready_buffer_1 == ready_buffer_2`?

Reply:
`ItemWithReadyKey` implements `PartialEq` so that it can implement `Ord` so that it can be used in a `BinaryHeap` within `ReadyBuffer`. `ReadyBuffer`'s comments describe its functions as dependent on the order of the key (e.g. a tick or instant) and does not mention being dependent on the order of the item's value.

`ReadyBuffer` doesn't implement `PartialEq` so `ready_buffer_1 == ready_buffer_2` doesn't compile but if it did then the equality would describe whether or not the two buffers have the same number of keys.